File: ScratchSnakeNode/project.js
const axios = require('axios');

async function request(url, err) {
  try {
    const response = await axios.get(
      url,
    );
    return response.data;
  }
  catch (error) {
    console.error(
      err,
    );
    return null;
  }
}

class Project {
  constructor(id) {
    this.id = id;
  }

  async update() {
    this.data = await request(
      `https://api.scratch.mit.edu/projects/${this.id}`,
      `ERROR: couldn't fetch https://api.scratch.mit.edu/projects/${this.id}.`
    )
    try {
      this.raw_json = this.data;
      this.title = this.data["title"];
      this.instructions = this.data["instructions"];
      this.notes = this.data["description"];
      this.comments = this.data["comments_allowed"];
      this.author = this.data["author"]["username"];
      this.logo = this.data["image"];
      this.created = this.data["history"]["created"];
      this.modified = this.data["history"]["modified"];
      this.shared = this.data["history"]["shared"];
      this.views = this.data["stats"]["views"];
      this.loves = this.data["stats"]["loves"];
      this.favs = this.data["stats"]["favorites"];
      this.remixes = this.data["stats"]["remixes"];
      this.token = this.data["project_token"];
    }
    catch (error) {
      this.raw_json = null;
      this.title = null;
      this.instructions = null;
      this.notes = null;
      this.comments = null;
      this.author = null;
      this.logo = null;
      this.created = null;
      this.modified = null;
      this.shared = null;
      this.views = null;
      this.loves = null;
      this.favs = null;
      this.remixes = null;
      this.token = null;
    }
  }
}

module.exports = { Project };

File: ScratchSnakePy/project.py
import requests as req

class Project:
    def get(self, project_id):
        self.project = project_id
        try:
            self.project_data = req.get(f"https://api.scratch.mit.edu/projects/{self.project}").json()
        except Exception:
            print(f"ERROR: couldn't fetch https://api.scratch.mit.edu/projects/{self.project}.")
    
    def json(self):
        return self.project_data
    
    def title(self):
        return self.project_data["title"]
    
    def instructions(self):
        return self.project_data["instructions"]
    
    def notes(self):
        return self.project_data["description"]
    
    def comments(self):
        return self.project_data["comments_allowed"]
    
    def author(self):
        return self.project_data["author"]["username"]
    
    def logo(self):
        return self.project_data["image"]
    
    def created(self):
        return self.project_data["history"]["created"]
    
    def modified(self):
        return self.project_data["history"]["modified"]
    
    def shared(self):
        return self.project_data["history"]["shared"]
    
    def views(self):
        return self.project_data["stats"]["views"]
    
    def loves(self):
        return self.project_data["stats"]["loves"]
    
    def favs(self):
        return self.project_data["stats"]["favorites"]
    
    def remixes(self):
        return self.project_data["stats"]["remixes"]
    
    def token(self):
        return self.project_data["project_token"]